helpers/hash.js

```diff
@@ -0,0 +1,12 @@
+const crypto = require('crypto');
+
+module.exports = {
+    /**
+     * @type {function(URL):string}
+     */
+    createUrlHash: url => {
```

Comment:
nit: the result of this function compared to it's name is unexpected IMO. How about we change the name to `createFileNameFromUrl`?